common/ext/class.Namespace.php

```diff
@@ -40,130 +40,81 @@ class common_ext_Namespace
      * @access protected
      * @var int
      */
-    protected $modelId = 0;
+    protected $modelId;
 
     /**
      * the namespace URI
      *
      * @access protected
      * @var string
      */
-    protected $uri = '';
+    protected $uri;
 
     // --- OPERATIONS ---
 
     /**
      * Create a namespace instance
      *
-     * @access public
-     * @author Jerome Bogaerts, <jerome.bogaerts@tudor.lu>
-     * @param  int id
-     * @param  string uri
-     * @return mixed
+     * @param string|int id
+     * @param string uri
      */
-    public function __construct($id = 0, $uri = '')
+    public function __construct($id, $uri = '')
     {
-        
-        
-    	if($id > 0){
-    		$this->modelId = $id;
-    	}
-    	if(!empty($uri)){
-    		$this->uri = $uri;
-    	}
-    	
-        
+  		$this->modelId = $id;

```

Comment:
should we check if `empty($id)` and use a uuid4 generator to have as default value ? 
also by removing the  check `if(!empty($uri)){` it can mean that the $uri can have the value `null`

Reply:
It's only used for creating Namespace objects retrieved from db, so with either integer or strings as $id. It must not be called with null or no value...